app/views/layouts/base.html.erb

```diff
@@ -23,6 +23,9 @@
     <%= APP_NAME %><%= raw(" - #{yield(:title)}") if content_for?(:title) %>
   </title>
 
+  <%= nonced_javascript_tag do %>
+    document.documentElement.className = document.documentElement.className.replace(/\bno-js\b/, 'js');
```

Comment:
this is because we'd have to polyfill `classList` in order to do `classList.toggle()`?

Reply:
> this is because we'd have to polyfill `classList` in order to do `classList.toggle()`?

Not so much that, since we're doing a swap from `no-js` to `js`, so an equivalent `classList` implementation wouldn't be all that much more concise:

```js
document.documentElement.classList.remove('no-js');
document.documentElement.classList.add('js');
```

And yes, we can't assume any polyfills here, and loading polyfills would defeat most of the purpose of pulling this out into the base layout, which is to make sure that it's run as early as possible to avoid a flash of content.